include/ginkgo/core/preconditioner/isai.hpp

```diff
@@ -143,6 +164,25 @@ class Isai : public EnableLinOp<Isai<IsaiType, ValueType, IndexType>>,
          * Must be at least 1, default value 1.
          */
         int GKO_FACTORY_PARAMETER_SCALAR(sparsity_power, 1);
+
+        /**
+         * @brief Size limit for the excess system.
+         *
+         * For rows with more than 32 nonzero entries, the algorithm builds up
+         * an excess system which is solved with sparse triangular solves (for
+         * upper or lower ISAI) or GMRES (for general ISAI). If this parameter
+         * is set to some m > 0, the excess system is solved as soon
+         * as its size supersedes m. This is repeated until the complete excess
+         * solution has been computed.
+         * Must be at least 0, default value 0.
+         */
+        size_type GKO_FACTORY_PARAMETER_SCALAR(excess_limit, 0u);
+
+        /**
+         * Factory for the Excess System solver
```

Comment:
Could you also add the default solver for different cases if it is not set by user